### ISSUE TITLE
Improve configuration file generation 

### ISSUE THREAD

**Issue by JasonMoho** [marius-team/marius#9]
**Is your feature request related to a problem? Please describe.**
Currently configuration generation is performed for every call to preprocess.py and three configuration files are generated. One for CPU training, one for GPU training, and one for multi-GPU training.

**Describe the solution you'd like**
We should put this generation into a separate optional step where it can be included in the preprocessing by adding a flag to the preprocessor call. 

E.g.
```
python3 preprocess.py fb15k output_dir/                             // No config generated
python3 preprocess.py fb15k output_dir/ --generate_config           // generates a single-GPU training configuration file by default
python3 preprocess.py fb15k output_dir/ --generate_config=GPU       // generates a single-GPU training configuration file
python3 preprocess.py fb15k output_dir/ --generate_config=CPU       // generates a CPU training configuration file
python3 preprocess.py fb15k output_dir/ --generate_config=Multi-GPU // generates a multi-GPU training configuration default
```

Adding config arguments should be supported too.

```
python3 preprocess.py fb15k output_dir/ --generate_config=GPU  <args>
python3 preprocess.py fb15k output_dir/ --generate_config=GPU  --model.embedding_size=400     // generates a single-GPU training configuration file for 400 dimensional embeddings
```

We should also allow for this configuration generation to be called separately. E.g.

```
python3 generate_config.py <files> <args> // args should include --embedding_dimension --num_partitions and --config_type (GPU, CPU, multi-GPU)
```

**Describe alternatives you've considered**
We could also disable this feature, but providing configuration file generation makes it easier on the users to get up and running on built-in and custom datasets.

**Additional context**
Eventually we will want to support generating a config based on user system characteristics. E.g. The users system has 64 GB of memory and wants to train 400 dimension embeddings on Freebase86m. We can set the number of partitions and buffer capacity to well utilize the available memory.

**Comment by shivaram** [comment]
One more thing is that in the future `generate_config.py` could try to automatically determine num partitions, batch size etc.  Also I do think that having separate utilities for both would be better than having one utility that does two things

**Comment by JasonMoho** [comment]
Fixed with #14

**Comment by JasonMoho** [comment]
Fully implemented in #20
